clients/v1_repl/v1_repl.go

```diff
@@ -0,0 +1,912 @@
+		c.setFormat(cmdArgs)
+	case "precision":
+		c.setPrecision(cmdArgs)
+	case "settings":
+		c.settings()
+	case "pretty":
+		c.togglePretty()
+	case "use":
+		c.use(cmdArgs)
+	case "insert":
+		c.insert(cmd)
+	case "clear":
+		c.clear(cmd)
+	default:
+		c.runAndShowQuery(cmd)
+	}
+}
+
+// Create a regex string for a named InfluxQL identifier, quoted or unquoted
+func identRegex(name string) string {
+	return `((?P<` + name + `>\w+)|(\"(?P<` + name + `_quote>.+)\"))`
+}
+
+// Get the value of a named InfluxQL identifier from a regex match map.
+// Returns empty string if no match
+func getIdentFromMatches(matches *map[string]string, name string) string {
+	if val, ok := (*matches)[name]; ok && val != "" {
+		return val
+	} else if val, ok := (*matches)[name+"_quote"]; ok && val != "" {
+		return val
+	}
+	return ""
+}
+
+// Create a regex match map from a regexp with named groups.
+// Returns nil if no match.
+func reSubMatchMap(r *regexp.Regexp, str string) *map[string]string {
+	match := r.FindStringSubmatch(str)
+	if match == nil {
+		return nil
+	}
+	subMatchMap := make(map[string]string)
+	for i, name := range r.SubexpNames() {
+		if i != 0 {
+			subMatchMap[name] = match[i]
+		}
+	}
+	return &subMatchMap
+}
+
+var insertIntoRegex string = `^(?i)INSERT INTO ` + identRegex("db") + `(\.` + identRegex("rp") + `)? (?P<point>.+)$`
+var insertRegex string = `^(?i)INSERT (?P<point>.+)$`
+
+func (c Client) insert(cmd string) {
+	var db string
+	var rp string
+	var point string
+	insertRgx := regexp.MustCompile(insertRegex)
+	insertIntoRgx := regexp.MustCompile(insertIntoRegex)
+	insertMatches := reSubMatchMap(insertRgx, cmd)
+	insertIntoMatches := reSubMatchMap(insertIntoRgx, cmd)
+	if insertIntoMatches != nil {
+		db = getIdentFromMatches(insertIntoMatches, "db")
+		rp = getIdentFromMatches(insertIntoMatches, "rp")
+		point = getIdentFromMatches(insertIntoMatches, "point")
+		if db != "" && rp == "" {
+			defaultRp, err := c.getDefaultRetentionPolicy(context.Background(), db)
+			if err != nil {
+				color.Red("Unable to get default retention policy for %q", db)
+				return
+			}
+			rp = defaultRp
+		}
+	} else if !strings.HasPrefix(strings.ToUpper(cmd), "INSERT INTO") && insertMatches != nil {
+		db = c.Database
+		rp = c.RetentionPolicy
+		point = getIdentFromMatches(insertMatches, "point")
+		if db == "" {
+			color.Red("Please run \"use <database>\" to run \"INSERT <point\"")
+			return
+		}
+	} else {
+		color.Red("Expected \"INSERT INTO <database>.<retention_policy> <point>\" OR \"INSERT <point>\".")
+		return
+	}
+
+	buf := bytes.Buffer{}
+	gzw := gzip.NewWriter(&buf)
+
+	_, err := gzw.Write([]byte(point))
+	gzw.Close()
+	if err != nil {
+		color.Red("Failed to gzip points")
+		return
+	}
+	bucketID, err := c.getBucketIdForDBRP(db, rp)
+	if err != nil {
+		color.Red("Unable to match DBRP to BucketID: %v", err)
+		return
+	}
+	writeReq := c.PostWrite(context.Background()).
+		Bucket(bucketID).
+		Precision(api.WritePrecision(c.Precision)).
```

Comment:
@lesam Looking at this api.WritePrecision, it only expects from `{"ms", "s", "us", "ns"}`.
Should this be using a new`c` field, like `c.WritePrecision` to write, since we want to support `rfc3339` precisions on queries?

Reply:
It allowed you to do that, then I think the query engine would give you an error.
I've added the option for `rfc3339` precision, and when you run an `INSERT` command, this error comes back:
`ERR: 400 Bad Request: invalid precision; valid precision units are ns, us, ms, and s`. Is this acceptable or should I do checking/clarification beforehand?

Reply:
Hmm - looks like it 'just worked' in 1.x, and rfc3339 on query meant `ns` on write:

```
> precision rfc3339
> select * from cpu limit 10
name: cpu
time                 cpu       host              usage_guest usage_guest_nice usage_idle        usage_iowait usage_irq usage_nice usage_softirq usage_steal usage_system        usage_user
----                 ---       ----              ----------- ---------------- ----------        ------------ --------- ---------- ------------- ----------- ------------        ----------
2022-04-28T20:30:20Z cpu-total MacBook-Pro.local 0           0                96.36120847989224 0            0         0          0             0           1.49390752796169    2.1448839926075656
2022-04-28T20:30:20Z cpu0      MacBook-Pro.local 0           0                84.85456369129106 0            0         0          0             0           6.218655967875164   8.926780340833778
2022-04-28T20:30:20Z cpu1      MacBook-Pro.local 0           0                99.8997995991049  0            0         0          0             0           0                   0.10020040080397738
2022-04-28T20:30:20Z cpu10     MacBook-Pro.local 0           0                97.79779779802409 0            0         0          0             0           0.9009009009241942  1.3013013013430397
2022-04-28T20:30:20Z cpu11     MacBook-Pro.local 0           0                99.79999999981374 0            0         0          0             0           0.10000000000218279 0.09999999999763531
2022-04-28T20:30:20Z cpu2      MacBook-Pro.local 0           0                89.6896896894004  0            0         0          0             0           4.504504504479571   5.805805805828698
2022-04-28T20:30:20Z cpu3      MacBook-Pro.local 0           0                99.89989989980658 0            0         0          0             0           0                   0.10010010009782637
2022-04-28T20:30:20Z cpu4      MacBook-Pro.local 0           0                92.29229229209865 0            0         0          0             0           3.1031031030826894  4.604604604600158
2022-04-28T20:30:20Z cpu5      MacBook-Pro.local 0           0                99.89979959910723 0            0         0          0             0           0.10020040079936143 0
2022-04-28T20:30:20Z cpu6      MacBook-Pro.local 0           0                95.29058116037271 0            0         0          0             0           1.8036072143930624  2.905811623119968
> insert m0,tag1=foo bval=asdf 1640775019000000000
ERR: {"error":"unable to parse 'm0,tag1=foo bval=asdf 1640775019000000000': invalid boolean"}

> insert m0,tag1=foo bval=1.2 1640775019000000000
> select * from m0
name: m0
time                 bval tag1
----                 ---- ----
2021-12-29T10:50:19Z 1.2  foo
```

Having it as an error for now seems fine - @samhld do you want a follow-up for something like a 'write precision' separate from read precision?